resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ClientSuite.scala

```diff
@@ -433,4 +465,36 @@ class ClientSuite extends SparkFunSuite with Matchers {
     classpath(env)
   }
 
+  private def testResourceRequest(expectedResources: Seq[(String, Long)],
```

Comment:
This is weird. The expected resources are a parameter, but the config is hardcoded in this test method. That is kinda brittle.

Instead, use the `deployMode` parameter to define the expected resources.

Reply:
This isn't fixed yet. You're still passing in the expected resources, which are expected to match the hardcoded configs that are set in this method.